crypto/fipsmodule/ec/oct.c

```diff
@@ -118,7 +125,10 @@ size_t ec_point_to_bytes(const EC_GROUP *group, const EC_AFFINE *point,
 int ec_point_from_uncompressed(const EC_GROUP *group, EC_AFFINE *out,
                                const uint8_t *in, size_t len) {
   const size_t field_len = BN_num_bytes(&group->field.N);
-  if (len != 1 + 2 * field_len || in[0] != POINT_CONVERSION_UNCOMPRESSED) {
+  // |POINT_CONVERSION_HYBRID| has the solution of y encoded in the first byte
+  // as well.
+  if (len != 1 + 2 * field_len || (in[0] != POINT_CONVERSION_UNCOMPRESSED &&
```

Comment:
why do we handle the hybrid case in function called `ec_point_from_uncompressed`?

Reply:
When we ignore the first byte (which reveals the encoding format), uncompressed form is essentially identical to hybrid. It seemed like a bit redundant to write another `ec_point_from_hybrid` function with `uncompressed` parsing logic.
I could rename this function for a bit more clarity if needed? I was also considering adding a check in the hybrid decoding to see if `y` actually had the same solution that the first byte indicated, is that something we should do?

Reply:
I'd prefer having a separate `ec_point_from_hybrid` function, it's redundant but simpler.

> I was also considering adding a check in the hybrid decoding to see if y actually had the same solution that the first byte indicated, is that something we should do?

That's a great question. Since we are doing this for compatibility with OpenSSL I guess you should see if they are checking the point consistency?

Reply:
Sure, I did a quick look into what OpenSSL was doing. They have [a check](https://github.com/openssl/openssl/blob/8ff6edb9da6199b130bfb50bc27b2e58cc815932/crypto/ec/ecp_oct.c#L351-L352) for consistency that goes both ways. 
Our's will look slightly different since we save the date in `EC_FELEM`, but I'll figure something out here.